bin/auth/insert_tokens.py

```diff
@@ -0,0 +1,25 @@
+import argparse
+import uuid
+import emission.storage.decorations.token_queries as esdt
+
+def get_tokens_from_file(file):
+    with open(file) as tlf:
+        raw_token_list = tlf.readlines()
+    token_list = [t.strip() for t in raw_token_list]
+    return(token_list)
+
+def load_tokens_into_db(token_list, userid):
+    for t in token_list:
+        esdt.insert({'user_id':userid,"token":t})
```

Comment:
Why do we need to associate a `user_id` with the token, particularly given that the `user_id`, as passed in, is a single value for the entire list?

Reply:
This was something motivated by following examples I saw in the tests. The user_id is used to delete DB entries between tests, is what I thought I noticed. I don't know everything that goes on in the unit tests, I assume that they are meant to be kept as independent and contained as possible, but if there was reason to keep some things in the DB between tests, you needed to clear out some but not the others, and it looked like that was done via lines like `edb.get_analysis_timeseries_db().delete_many({'user_id': self.testUserId})`. At some point I changed my test to just `delete_many({})`, I don't remember when, and it worked fine, but I forgot to do away with the user_ids everywhere else.

I followed that pattern when building out the functions, and when I went to write the script, I think I just saw that there was room to provide this, so I put it in without thinking too much about it. I wasn't even sure I needed it when following the test examples in the first place, I can easily get rid of it.

Reply:
I see. In general, the data in the timeseries and analysis timeseries collection are tied to an individual who is represented by the UUID.  That is why we insert for a particular UUID and then delete for the same UUID. In the case of the tokens, I don't see a specific need for them to be associated with a user.

In general, I would prefer for the DB design to not be driven solely by the unit tests; as you pointed out, you are already deleting everything `delete_many({})` anyway. If you wanted to be a bit more careful about that, so that inadvertently running the tests against an existing database would not break it, you could make sure to only delete the values that you insert.

This would have the same function as the current UUID-based deletion for the other unit tests.